classes/plagiarism_copyleaks_eventshandler.class.php

```diff
@@ -592,4 +626,56 @@ public function handle_eula_acceptance($data, $isretry = false) {
             }
         }
     }
+
+
+    /**
+     * get assign submission data
+     * @param object $data
+     * @param object $coursemodule
+     * @param string $authoruserid
+     * @param string $submitteruserid
+     * @param object $cmdata
+     * @param object $identifier
+     * @param string $subtype
+     * @param int $scheduledscandate
+     */
+    private function get_assign_submission_data($itemid, $coursemodule, $authoruserid, $cmdata, $identifier, $subtype, $scheduledscandate) {
+        global $DB;
+
+
+        $submissiondata = array();
+        $submissionrecord = $DB->get_record(
+            'assign_submission',
```

Comment:
@ShadeAma  Working directly with the assign submission table is a bad practice. Why not use the [Submissions Library API](https://docs.moodle.org/dev/Submissions_Library)?

Reply:
I couldn’t find an alternative to retrieving the data without directly accessing the assign_submission table. This approach is also used multiple times in the older code within plagiarism_copyleaks_eventshandler.class.php, making it consistent with existing practices.